tests/unit/experiments/providers/simple_experiment_tests.py

```diff
@@ -354,3 +377,45 @@ def test_change_shuffle_version_changes_bucketing(self):
                 break
 
         self.assertTrue(bucketing_changed)
+
+    def test_generate_targeting_valid_config(self):
+        targeting_cfg = get_targeting_config()
+
+        experiment_correct_targeting = _generate_targeting(targeting_cfg)
+        self.assertTrue(isinstance(experiment_correct_targeting, Targeting))
+
+        self.assertFalse(isinstance(experiment_correct_targeting, OverrideNode))
+
+    def test_generate_targeting_no_config(self):
+        targeting_cfg = None
+
+        experiment_correct_targeting = _generate_targeting(targeting_cfg)
+        self.assertTrue(isinstance(experiment_correct_targeting, Targeting))
+
+        self.assertTrue(isinstance(experiment_correct_targeting, OverrideNode))
+        self.assertTrue(experiment_correct_targeting.evaluate())
+
+    def test_generate_targeting_invalid_config(self):
+        targeting_cfg = get_targeting_config()
+        targeting_cfg['ALL'][0]['ANY'][0] = {'EQUAL': {'field': 'is_mod', 'value': True}}
+
+        experiment_correct_targeting = _generate_targeting(targeting_cfg)
+        self.assertTrue(isinstance(experiment_correct_targeting, Targeting))
+
+        self.assertTrue(isinstance(experiment_correct_targeting, OverrideNode))
+        self.assertFalse(experiment_correct_targeting.evaluate())
+
+    def test_targeting_in_config(self):
+        cfg = get_simple_config()
+        targeting_cfg = get_targeting_config()
+        cfg['experiment']['targeting'] = targeting_cfg
+
+        experiment_with_targeting = parse_experiment(cfg)
+
+        experiment_inputs = {
+            'is_mod':True,
+            'is_logged_in':True,
+            'random_numeric':5,
+        }
+
+        self.assertTrue(experiment_with_targeting.is_targeted(**experiment_inputs))
```

Comment:
🔕 maybe style nit:  This seems simple enough to just set the kwargs directly rather than using a dict and the `**` syntax.  I feel like that makes it more clear what the call args actually are.  If other's feel differently or don't feel strongly about it, then this is fine too.

Reply:
yes please